src/features/deleteUser/model/store.ts

```diff
@@ -0,0 +1,14 @@
+import { sample, createEvent } from 'effector';
+import { viewerModel } from 'entities/viewer';
+
+export const userDeleted = createEvent();
+
+sample({
+  clock: userDeleted,
+  source: viewerModel.$user,
+  fn: (user) => {
+    if (!user) throw new Error('Failed to delete User');
```

Comment:
not sure we should handle it

Reply:
if it's about fn - I deleted fn. if about source - without source: 'viewerModel.$user'  will be a problem with TS.